firebase-sessions/src/main/kotlin/com/google/firebase/sessions/SessionGenerator.kt

```diff
@@ -0,0 +1,74 @@
+/*
+ * Copyright 2023 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.google.firebase.sessions
+
+import java.util.UUID
+
+/**
+ * [SessionInfo] is a data class responsible for storing information about the current Session.
+ *
+ * @hide
+ */
+internal data class SessionInfo(
+  val sessionId: String,
+  val firstSessionId: String,
+  val collectEvents: Boolean,
+  val sessionIndex: Int,
+)
+
+/**
+ * The [SessionGenerator] is responsible for generating the Session ID, and keeping the
+ * [SessionInfo] up to date with the latest values.
+ *
+ * @hide
+ */
+internal class SessionGenerator(collectEvents: Boolean) {
+  private var firstSessionId = ""
+  private var sessionIndex: Int = -1
+  private var collectEvents = collectEvents
```

Comment:
Do we plan to have a dynamic toggle of `collectEvents` state? Eg: SessionGenerator starts with the collectEvents state to be `false` and somewhere between turns to be `true`. Is this a usecase we want to support?

Reply:
Not anymore - we decided that the collectEvents state is consistent for the run of the app